src/runtime/components/prose/Callout.vue

```diff
@@ -1,15 +1,16 @@
 <script lang="ts">
+import type { Component } from 'vue'
 import type { AppConfig } from '@nuxt/schema'
 import theme from '#build/ui/prose/callout'
-import type { LinkProps } from '../../types'
+import type { IconProps, LinkProps } from '../../types'
 import type { ComponentConfig } from '../../types/tv'
 
 type ProseCallout = ComponentConfig<typeof theme, AppConfig, 'callout', 'ui.prose'>
 
 export interface ProseCalloutProps {
   to?: LinkProps['to']
   target?: LinkProps['target']
-  icon?: string
+  icon?: Exclude<IconProps['name'], Component>
```

Comment:
I'd just keep `IconProps['name']` here as the component can be used as a Vue component:
```suggestion
  icon?: IconProps['name']
```